### PR TITLE
Story 3

### PR DESCRIPTION
**Pull request by agallant121**
Complete user story 3

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
 class ApplicationController < ActionController::Base
+  before_action :current_user
```
Comment (agallant121) [inline app/controllers/application_controller.rb:2]: Add current user for the before_action to find out if the current user exists

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :current_user
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def current_user
```
Comment (agallant121) [inline app/controllers/application_controller.rb:11]: Add current_user helper method to be called throughout application

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ def create
     # @user = User.new(user_params)
     @user = User.create(user_params)
     if @user.save
+      session[:user_id] = @user.id
```
Comment (agallant121) [inline app/controllers/users_controller.rb:11]: Add session creation for new users

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -15,8 +15,14 @@
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
       <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register as New User", "/register" %>
-      <%= link_to "Login", "/login" %>
+      <% if current_user %>
+        <%= link_to "My Profile", "/profile" %>
```
Comment (agallant121) [inline app/views/layouts/application.html.erb:19]: Adding conditionals to show certain things within the nav bar depending on if the user is logged in or not

File: spec/features/navigation/navigation_spec.rb [hunk 5]
```diff
@@ -1,22 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe 'Site Navigation' do
-  before :each do
 
-    visit '/'
+  it 'shows this information from the nav bar' do
+  visit '/'
+      within('#top-nav') do
```
Comment (agallant121) [inline spec/features/navigation/navigation_spec.rb:7]: Edit first test to be able to see stuff from nav bar as a regular, not-logged-in, visitor

File: spec/features/navigation/navigation_spec.rb [hunk 6]
```diff
@@ -28,31 +41,25 @@
...
-        expect(page).to have_link("Home")
-        expect(page).to have_link("All Merchants")
-        expect(page).to have_link("All Items")
-        expect(page).to have_link("Cart")
-        expect(page).to have_link("Login")
-        expect(page).to have_link("Register as New User")
-        expect(page).to have_content("Cart: 0")
-    end
+    expect(page).to have_content("Logged in as Happy Gilmore")
+    expect(page).to have_link("My Profile")
```
Comment (agallant121) [inline spec/features/navigation/navigation_spec.rb:46]: Story three tests looking for changes in navbar when logged in

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (palworth) [reply app/controllers/users_controller.rb:11]: initializing session

**Comment by palworth** [comment]
US 3 complete. Ready for master.

**Pull request merged**